code/game/objects/items.dm

```diff
@@ -487,9 +485,11 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
 		if(!do_after(user, 30*grav_power, src))
 			return
 
-
 	//If the item is in a storage item, take it out
-	SEND_SIGNAL(loc, COMSIG_TRY_STORAGE_TAKE, src, user.loc, TRUE)
+	if(loc.GetComponent(/datum/component/storage))
+		if(SEND_SIGNAL(loc, COMSIG_TRY_STORAGE_TAKE, src, user, TRUE)) // this calls 'remove_from_storage()' proc
+			return
```

Comment:
Checking for a specific component before using a signal that returns a value if successful makes no sense